### PR TITLE
Retry when attempting to connect to Franka for first time

### PR DESCRIPTION
**Pull request by 5yler**
### Background

- This prevents us from bailing early and creating unnecessary duplicate logs each time driver fails to start. Just keep trying...

### What's new
- ✅ [New feature description. Only a single new major feature is allowed per PR.]
- ✅❌ New feature is accompanied by new test: [name and description of new test].

### Tests
1. ✅Tested on simulated robot: [Describe the tests/commands used.]
2. ✅ Tested on physical robot: Tried starting up while user stopped or not connected to franka, verified it keeps retrying until connection can be established

### Quality
3. ✅ New code is written in pure C++17 to the best of my knowledge.
4. ✅ New code follows established C++17 best practices ([C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines)).
5. ✅ New code passed `clang-format` and `cpplint` at least, if not all of our static code analysers.
6. ✅ I have included Doxygen-style documentation in the new C++ code.

### Note to reviewers
Please verify that all sections are accurately filled out and that all 6 numbered entries above are present and ticked/checked off, with their requirements met, if applicable.

### CODE REVIEW

File: src/franka_plan_runner.cc [hunk 1]
```diff
@@ -155,48 +155,45 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
...
-          utils::RobotModeToString(current_mode));
-      try {
-        robot.automaticErrorRecovery();
-        dexai::log()->info(
-            "RunFranka: automaticErrorRecovery() succeeded, "
-            "robot now in mode: {}.",
+  bool connection_established {false};
+  // attempt connection to robot and read current mode
+  // and if it fails, keep trying instead of exiting the program
+  while (!connection_established) {
```
Comment (aneksteind) [inline src/franka_plan_runner.cc:161]: should we have a sleep in here just in case? don't want another CPU hungry driver

Comment (5yler) [reply src/franka_plan_runner.cc:161]: Fair point. I'll add it in

File: src/franka_plan_runner.cc [hunk 2]
```diff
@@ -155,48 +155,51 @@ franka::RobotMode FrankaPlanRunner::GetRobotMode(franka::Robot& robot) {
...
+    try {
+      franka::Robot robot(ip_addr_);
+
+      auto current_mode = GetRobotMode(robot);
```
Comment (duanyutong) [inline src/franka_plan_runner.cc:165]: uniform init

**Review by duanyutong — APPROVED** [review]
lgtm

**Review by aneksteind — APPROVED** [review]

**Pull request merged**